qiskit_nature/operators/second_quantization/vibrational_spin_op_utils/vibr_to_spin_op_label_converter.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+This module converts labels from the `VibrationalSpinOp` to 'SpinOp` notation.
+"""
+import itertools
+import operator
+import re
+from typing import List, Tuple, Union
+
+from qiskit_nature.operators.second_quantization.vibrational_spin_op_utils\
+    .vibrational_labels_validator import \
+    validate_vibrational_labels
+
+
+def convert_to_spin_op_labels(vibrational_labels: List[Tuple[str, complex]], num_modes: int,
```

Comment:
This is some separate module, and public supported API, outside of the VibrationalSpinOp class as its considered useful in its own right. I could see this as function in the VibrationalSpinOp - just wondering why we need this and the others in this folder that seem more like private helper functions of that class

Reply:
Update, I see there was a conversation above about this - sorry I have been trying to catch up with things and did not see that first!

Reply:
I can see your point. My main motivation was to extract logic with a certain responsibility from the main class. As you can see these modules end up being pretty lengthy. Then, I have been wondering if they could be useful to a user in their own right in case they do some steps on their own - but I am not convinced about it. I would be fine with making all methods in these modules "private" such that they are not exposed as a public API. Let me know if there is a better way to accomplish it and avoid importing a private method to VibrationalSpinOp.

Reply:
I guess personally I am less worried about the resultant total length of VibrationalSpinOp if these are added at the end of it. The methods themselves are reasonable and so reading them in the overall file should be fine. It would only end up being 2 ro 3 hundred lines long right - puny compared to say the 2.6K that is QuantumCircuit!